move TLS response cache invalidation to common section

fmauchle opened the pull request:
Invalidation of response cache applies to both TLS and DTLS since both are TLS session based. Move paragraph to the common section.

alandekok commented:
```
Since the client connection is closed, those responses from the home server to the proxy server SHOULD be silently discarded by the proxy
```

Maybe make this MUST?  If the home server replies, there's no way for the proxy to send the packet anywhere.

fmauchle commented:
> Maybe make this MUST? If the home server replies, there's no way for the proxy to send the packet anywhere.

Agreed; updated.

Janfred commented:
I would change the title of this section.
It's not limited to cached response packets, it's also outstanding requests and all the other stuff that can happen if a connection gets torn down.
I'll try to suggest a different title for this soon.

On another note: This again is something that is not necessarily specific to RADIUS/(D)TLS, it only comes up since we now have the concept of "sessions", where in RADIUS/UDP the proxy would just forward the response to the original client, despite it being down (since UDP does not have the concept of sessions, so how would the proxy know that the client is gone?)

alandekok commented:
+1.  It's _all_ responses.  I'll also note the text in https://github.com/radext-wg/draft-ietf-radext-radiusdtls-bis/pull/6/commits/6c10a64b51930fd027cff1bf1405aeb1ac30ed2b doesn't make it clear that this is for _incoming_ connections

i.e.
* in an incoming connection is closed, then there is no way to send responses to anu requests which were received on that connection.  As a result, all requests MUST be discarded / stopped processing.  etc.
* if an outgoing connection is closed, the proxy or client SHOULD try to send the packet over another outgoing connection.  Further details are outside of the scope of this specification.

We have this issue with TCP, too.  But that can be ignored.

For UDP, we have no way of knowing if the client is gone, so the server just always replies.

fmauchle commented:
Very good point. I will try to come up with an updated text.

fmauchle commented:
Updated proposed text.

I've also added a sentence that associated EAP MAY continue with a RAIDUS request on a new connection. @alandekok please correct me if this is not true. The underlying EAP session will (or at least can) also do retransmissions, sent in a completely new RADIUS request?

alandekok commented:
After thinking some more, there are more corner cases.  Perhaps:

If an incoming (D)TLS session or the underlying connection is closed or broken, then there is no way to send any response to the client.  The server behavior then depends on the types of packets being processed, and on the behavior of the server.

A proxy which forwards packet MUST discard all requests associated with the closed connection.  As no response can be sent over the (D)TLS connection, and any further processing of requests is pointless.  A discarded request may have a cached RADIUS response packet ({{!RFC5080, Section 2.2.2}}), in which case the cached response is also discarded.  If there is no cached response packet, then the request is still being processed.  The server SHOULD immediately stop processing all such requests for this incoming connection.

A home server which receives Access-Request packets MUST behave as defined above for a proxy.  Where a RADIUS packet is part of a multi-packet authentication session (e.g. EAP), the underlying authentication session could be continued, or the underlying authentication session data could be discarded.  The server may be able to receive and process another packet for that session via a different incoming connection.  It is difficult to make more recommendations for managing partially processed authentication sessions, as such recommendations depend strongly on the authentication method being used.  As a result, further behavior is implementation defined.

A home server which receives other kinds of packets (Accounting-Request, CoA-Request, Disconnect-Request) MAY finish processing outstanding requests, and then discard any response.  This behavior ensures that the desired action is still taken, even if the home server cannot inform the client of the results of that action.

Janfred commented:
I've moved the text to a new pull request ( #33 )

I'll mark this one as "wontfix", further discussion should happen in the new pull request
